src/components/TaskModal.js

```diff
@@ -0,0 +1,62 @@
+const getTaskModal = () => {
+  const modal = document.createElement('div');
+  modal.innerHTML = `
+    <div class="modal fade" id="taskModal" tabindex="-1">
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h5 class="modal-title">Add New Task</h5>
+            <button
+              type="button"
+              class="close"
+              data-dismiss="modal"
+              aria-label="Close"
+            >
+              <span aria-hidden="true">&times;</span>
+            </button>
+          </div>
+          <div class="modal-body">
+              <form>
+
+                <div class="form-group">
+                  <label for="task-name" class="col-form-label">Task:</label>
+                  <input type="text" class="form-control" id="task-name" pattern="^([_A-z0-9]){3,}$" required>
+                </div>
+
+                <div class="form-group">
+                  <label for="task-description-text" class="col-form-label">Description:</label>
+                  <textarea class="form-control" id="task-description"></textarea>
+                </div>
+
+                <div class="form-group">
+                 <label for="task-date" class="col-form-label">Due Date:</label>
+                 <input id='task-date' name='task-date' type="date" class="form-control" required>
+                 </div>
+                 
+                 <div class="form-group">
+                  <label for="task-priority" class="col-form-label">Priority:</label>
+                    <select name='task-priority' class="form-control" id="task-priority" required>
+                      <option value='Low'>Low</option>
+                      <option value='Medium'>Medium</option>
+                      <option value='High'>High</option>
```

Comment:
## Optional Change 
- [ ] Would you mind to add form validation\s helper message to help the user in identifying missing fields that need filling before a `task` gets created? This would go well with having a seamless User Experience when creating a `task ` for a project. thank you!

Reply:
Thanks @christianotieno for your feedback. I have addressed the issues you raised. Kindly request a pull request to merge the feature branch to the develop branch.

Thanks in advance.